### ISSUE TITLE
exception on maven-deployment

### ISSUE THREAD

**Issue by afischer211** [ctron/package-drone#110]
With release **0.10.0m4** I receive the following execption on uploading an artifact with Maven-Deployment:
```
javax.xml.transform.TransformerFactoryConfigurationError: Provider for class javax.xml.transform.TransformerFactory cannot be created
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:345) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder.find(FactoryFinder.java:299) ~[na:1.8.0_45]
        at javax.xml.transform.TransformerFactory.newInstance(TransformerFactory.java:106) ~[na:1.8.0_45]
        at de.dentrassi.pm.common.XmlHelper.<init>(XmlHelper.java:153) ~[de.dentrassi.pm.common_1.0.0.20150615-1109.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.processMetaData(MavenServlet.java:412) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.processPut(MavenServlet.java:267) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.doPut(MavenServlet.java:228) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:710) ~[javax.servlet-api_3.1.0.jar:3.1.0]
        at de.dentrassi.pm.storage.service.servlet.AbstractStorageServiceServlet.service(AbstractStorageServiceServlet.java:85) ~[de.dentrassi.pm.storage.servic                                                                                                                e_1.0.0.20150615-1109.jar:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet-api_3.1.0.jar:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808) ~[org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587) [org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70) [org.ops4j.pax.web.pax-web-jetty_4.1.2                                                                                                                .jar:na]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [org.eclipse.jetty.security_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:267) [org.ops4j.pax.web.pax-web-jetty_4.1.2.jar:na]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) [org.ops4j.pax.web.pax-web-jetty_4                                                                                                                .1.2.jar:na]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.Server.handle(Server.java:497) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [org.eclipse.jetty.io_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [org.eclipse.jetty.util_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [org.eclipse.jetty.util_9.2.9.v20150224.jar:9.2.9.v20150224]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45]
Caused by: java.lang.RuntimeException: Provider for class javax.xml.transform.TransformerFactory cannot be created
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:342) ~[na:1.8.0_45]
        ... 30 common frames omitted
Caused by: java.util.ServiceConfigurationError: javax.xml.transform.TransformerFactory: Provider org.apache.xalan.processor.TransformerFactoryImpl not found
        at java.util.ServiceLoader.fail(ServiceLoader.java:239) ~[na:1.8.0_45]
        at java.util.ServiceLoader.access$300(ServiceLoader.java:185) ~[na:1.8.0_45]
        at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:372) ~[na:1.8.0_45]
        at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404) ~[na:1.8.0_45]
        at java.util.ServiceLoader$1.next(ServiceLoader.java:480) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder$1.run(FactoryFinder.java:328) ~[na:1.8.0_45]
        at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:323) ~[na:1.8.0_45]
        ... 30 common frames omitted
11:58:32.872 [qtp1561382840-144] WARN  o.e.jetty.servlet.ServletHandler [ServletHandler.java:667] Error for /maven/<xxxxxxx-snipped-xxxxxxx>                                                                                                             /maven-metadata.xml
javax.xml.transform.TransformerFactoryConfigurationError: Provider for class javax.xml.transform.TransformerFactory cannot be created
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:345) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder.find(FactoryFinder.java:299) ~[na:1.8.0_45]
        at javax.xml.transform.TransformerFactory.newInstance(TransformerFactory.java:106) ~[na:1.8.0_45]
        at de.dentrassi.pm.common.XmlHelper.<init>(XmlHelper.java:153) ~[de.dentrassi.pm.common_1.0.0.20150615-1109.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.processMetaData(MavenServlet.java:412) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.processPut(MavenServlet.java:267) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at de.dentrassi.pm.maven.internal.MavenServlet.doPut(MavenServlet.java:228) ~[de.dentrassi.pm.maven_1.0.0.20150619-0749.jar:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:710) ~[javax.servlet-api_3.1.0.jar:3.1.0]
        at de.dentrassi.pm.storage.service.servlet.AbstractStorageServiceServlet.service(AbstractStorageServiceServlet.java:85) ~[de.dentrassi.pm.storage.servic                                                                                                                e_1.0.0.20150615-1109.jar:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet-api_3.1.0.jar:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808) ~[org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587) [org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70) [org.ops4j.pax.web.pax-web-jetty_4.1.2                                                                                                                .jar:na]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [org.eclipse.jetty.security_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:267) [org.ops4j.pax.web.pax-web-jetty_4.1.2.jar:na]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [org.eclipse.jetty.servlet_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) [org.ops4j.pax.web.pax-web-jetty_4                                                                                                                .1.2.jar:na]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.Server.handle(Server.java:497) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [org.eclipse.jetty.server_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [org.eclipse.jetty.io_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [org.eclipse.jetty.util_9.2.9.v20150224.jar:9.2.9.v20150224]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [org.eclipse.jetty.util_9.2.9.v20150224.jar:9.2.9.v20150224]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45]
Caused by: java.lang.RuntimeException: Provider for class javax.xml.transform.TransformerFactory cannot be created
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:342) ~[na:1.8.0_45]
        ... 30 common frames omitted
Caused by: java.util.ServiceConfigurationError: javax.xml.transform.TransformerFactory: Provider org.apache.xalan.processor.TransformerFactoryImpl not found
        at java.util.ServiceLoader.fail(ServiceLoader.java:239) ~[na:1.8.0_45]
        at java.util.ServiceLoader.access$300(ServiceLoader.java:185) ~[na:1.8.0_45]
        at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:372) ~[na:1.8.0_45]
        at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404) ~[na:1.8.0_45]
        at java.util.ServiceLoader$1.next(ServiceLoader.java:480) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder$1.run(FactoryFinder.java:328) ~[na:1.8.0_45]
        at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_45]
        at javax.xml.transform.FactoryFinder.findServiceProvider(FactoryFinder.java:323) ~[na:1.8.0_45]
        ... 30 common frames omitted

```

**Comment by ctron** [comment]
This is strange. Which JVM are you using?

**Comment by ctron** [comment]
Ok, I can reproduce it.

**Comment by ctron** [comment]
Ok, so here is what happens.

* The JBoss JSTL implementation requires Apache Xalan
* Apache Xalan brings in a service descriptor for file its XSLT processor
* This is read by the XmlHelper instantiated by the MavenServlet
* But cannot be found by it, since the MavenServlet does not require Xalan

**Comment by afischer211** [comment]
Hello, I use (like always) Oracle-Java8u45 (64bit) under CentOS7.

**Comment by afischer211** [comment]
This issue is solved with 0.10.0m5.
